consensus/poet/core/tests/test_consensus/test_poet_fork_resolver.py

```diff
@@ -0,0 +1,649 @@
+
+            # Could be a hack, but verify that the appropriate log message is
+            # generated - so we at least have some faith that the failure was
+            # because of what we are testing and not something else.  I know
+            # that this is fragile if the log message is changed, so would
+            # accept any suggestions on a better way to verify that the
+            # function fails for the reason we expect.
+
+            (message, *_), _ = mock_logger.info.call_args
+            self.assertTrue('New fork wait duration (%f) '
+                            'less than current fork wait duration '
+                            in message)
+
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                'ConsensusStateStore')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.factory')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                'ConsensusState')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                'ValidatorRegistryView')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.utils')
+    def test_different_previous_block_id(
+            self,
+            mock_utils,
+            mock_validator_registry_view,
+            mock_consensus_state,
+            mock_poet_enclave_factory,
+            mock_consensus_state_store,
+            mock_poet_config_view,
+            mock_block_wrapper):
+
+        """ When both current and new fork heads are valid
+            PoET blocks with different previous block ids,
+            the test verifies that the one with
+            the higher aggregate local mean wins
+        """
+
+        # create a mock_validator_registry_view
+        mock_validator_registry_view.return_value.get_validator_info. \
+            return_value = \
+            ValidatorInfo(
+                name='validator_001',
+                id='validator_deadbeef',
+                signup_info=SignUpInfo(
+                    poet_public_key='00112233445566778899aabbccddeeff'))
+
+        # create a mock_wait_certificate that does nothing in check_valid
+        mock_wait_certificate = mock.Mock()
+        mock_wait_certificate.check_valid.return_value = None
+        # set new_mock_wait_certificate local_mean and duration
+        mock_wait_certificate.local_mean = 0.0
+
+        # set mock_utils.deserialize_wait_certificate
+        # to return a specific value for each fork_head
+        # with cur_fork_head being deserialized first
+        mock_utils.deserialize_wait_certificate.side_effect = \
+            [mock_wait_certificate,
+             mock_wait_certificate,
+             mock_wait_certificate,
+             mock_wait_certificate,
+             mock_wait_certificate,
+             mock_wait_certificate,
+             mock_wait_certificate,
+             mock_wait_certificate,
+             mock_wait_certificate,
+             mock_wait_certificate
+             ]
```

Comment:
Since you are returning the same wait certificate multiple times, you may be able to simply:
`mock_utils.deserialize_wait_certificate.return_value = mock_wait_certificate`